root/io/io/TFileBenchmarks.cxx

```diff
@@ -0,0 +1,33 @@
+#include "TROOT.h"   // ROOT::EnableImplicitMT
+
+#include "TFile.h"	
+#include "TFrame.h"
+#include "TRandom3.h"
+#include "TBenchmark.h"
+
+#include "ROOT/RDataFrame.hxx"
+#include "ROOT/RSnapshotOptions.hxx"
+
+#include "benchmark/benchmark.h"
+#include "rootbench/RBConfig.h"
+
+
+static void BM_TFile_Write(benchmark::State &state) {
+	ROOT::EnableImplicitMT();
+	ROOT::RDF::RSnapshotOptions options;
+	options.fCompressionAlgorithm = ROOT::ECompressionAlgorithm::kZLIB;
+	//RCompressionSetting::EAlgorithm::EValues::kZLIB
+
+	for (auto _ : state) {
+		// We create an empty data frame
+	   ROOT::RDataFrame tdf(100000);
```

Comment:
Here you are benchmarking too many extra things, such as building RDataframe and filling it with random numbers, can we try to use fixture? What you actually need to benchmark in this case is Snapshot function https://root.cern/doc/master/classROOT_1_1RDF_1_1RInterface.html#a233b7723e498967f4340705d2c4db7f8.

Reply:
Yes, I forgot to move them out of the loop. What do you mean with fixture?

Reply:
Check about fixtures: https://github.com/google/benchmark#fixtures

Reply:
I am thinking how to apply it to this particular case. The example are a bit confusing if you are not familiar with the framework.
Should it be something like the following?

```
class MyFixture : public ::benchmark::Fixture {
public:
    void SetUp(const ::benchmark::State& st) {
       ROOT::EnableImplicitMT();
	   ROOT::RDF::RSnapshotOptions options;
	   options.fCompressionAlgorithm = ROOT::ECompressionAlgorithm::kLZ4;

	   // We create an empty data frame
	   ROOT::RDataFrame tdf(100000);
	   // We now fill it with random numbers
	   gRandom->SetSeed(1);
	   auto tdf_1 = tdf.Define("rnd", []() { return gRandom->Gaus(); });
	}

    void TearDown(const ::benchmark::State&) {}
    auto tdf_1;
};

BENCHMARK_F(MyFixture, BM_TFile_RDFSnapshot_ZLIB)(benchmark::State &state) {
	for (auto _ : state) {
		//And we write out the dataset on disk
		tdf_1.Snapshot("randomNumbers", "bench_data.root", "", options);
	}
}
```